engine/schema/src/com/cloud/vm/dao/NicDaoImpl.java

```diff
@@ -302,4 +309,17 @@ public int countNicsForStartingVms(long networkId) {
         List<Integer> results = customSearch(sc, null);
         return results.get(0);
     }
+
+    @Override
+    public Long getPeerRouterId(String publicMacAddress, final long routerId) {
+        final SearchCriteria<NicVO> sc = PeerRouterSearch.create();
+        sc.setParameters("instanceId", routerId);
+        sc.setParameters("macAddress", publicMacAddress);
+        sc.setParameters("vmType", VirtualMachine.Type.DomainRouter);
+        NicVO nicVo = findOneBy(sc);
+        if (nicVo != null) {
+            return (new Long(nicVo.getInstanceId()));
```

Comment:
Let the auto-boxing and auto-unboxing do this for you.
When you do this manually the pool of String/numbers of java will not be used.
You just need to return `nicVo.getInstanceId()`